### PR TITLE
Add an integration testing workflow

### PR DESCRIPTION
**Pull request by teqdruid**
Runs nightly or by request. Assisted by @stephenneuendorffer.

[Example build](https://github.com/llpm/circt/runs/1410544967):

image

### CODE REVIEW

File: .github/workflows/nightlyIntegrationTests.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            -DLLVM_ENABLE_PROJECTS='mlir' \
+            -DLLVM_OPTIMIZED_TABLEGEN=ON \
+            -DLLVM_ENABLE_OCAMLDOC=OFF \
+            -DLLVM_ENABLE_BINDINGS=OFF \
+            -DLLVM_INSTALL_UTILS=ON \
+            -DCMAKE_C_COMPILER=clang \
+            -DCMAKE_CXX_COMPILER=clang++ \
+            -DLLVM_ENABLE_LLD=ON \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DLLVM_ENABLE_ASSERTIONS=ON
```
Comment (stephenneuendorffer) [inline .github/workflows/nightlyIntegrationTests.yml:70]: This config keeps getting copied.  Any ideas for how to at least verify that we are getting the same configuration from the cache?

**Review by stephenneuendorffer — APPROVED** [review]

File: .github/workflows/nightlyIntegrationTests.yml (hunk 1 above)
Comment (teqdruid) [reply .github/workflows/nightlyIntegrationTests.yml:70]: Good point. Maybe move this into a script we store in the repo?

Comment (teqdruid) [reply .github/workflows/nightlyIntegrationTests.yml:70]: Looks like different workflows do not share caches. :(

Comment (teqdruid) [reply .github/workflows/nightlyIntegrationTests.yml:70]: Nonetheless, #267

**Pull request merged**